tajo-client/src/main/java/org/apache/tajo/client/SessionConnection.java

```diff
@@ -95,20 +96,29 @@ public SessionConnection(ServiceTracker tracker, @Nullable String baseDatabase,
     this.baseDatabase = baseDatabase != null ? baseDatabase : null;
 
     this.serviceTracker = tracker;
+    try {
+      this.client = getTajoMasterConnection();
+    } catch (Exception e) {
+      throw new IOException(e);
+    }
   }
 
   public Map<String, String> getClientSideSessionVars() {
     return Collections.unmodifiableMap(sessionVarsCache);
   }
 
-  public NettyClientBase getTajoMasterConnection(boolean asyncMode) throws NoSuchMethodException,
-      ConnectException, ClassNotFoundException {
-    return manager.getClient(getTajoMasterAddr(), TajoMasterClientProtocol.class, asyncMode);
-  }
+  public synchronized NettyClientBase getTajoMasterConnection() throws ServiceException {
```

Comment:
As I know, SessionConnection is created for each client. Do we need to declare synchronized for this method?